src/backend/marsha/core/api/video.py

```diff
@@ -611,47 +612,43 @@ class LiveRegistrationViewSet(
     def get_queryset(self):
         """Restrict access to liveRegistration with data contained in the JWT token.
 
-        Access is restricted to liveRegistration related to the video and context_id present in
-        the JWT token. Email or user id from the token can be used as well depending on the role.
+        Access is restricted to liveRegistration related to the video, context_id and consumer_site
+        present in the JWT token. Email is ignored. Lti user id from the token can be used as well
+        depending on the role.
         """
         user = self.request.user
 
-        if user.token.payload:
-            consumer_site = (
-                Playlist.objects.get(
-                    lti_id=user.token.payload["context_id"]
-                ).consumer_site
-                if user.token.payload.get("context_id")
-                else None
-            )
-            filters = {"consumer_site": consumer_site, "video__id": user.id}
+        is_lti = (
+            user.token.payload
+            and user.token.payload.get("context_id")
+            and user.token.payload.get("consumer_site")
+            and user.token.payload.get("user")
+            and user.token.payload["user"].get("id")
+        )
+
+        if is_lti:
+            filters = {"video__id": user.id}
             if self.kwargs.get("pk"):
                 filters["pk"] = self.kwargs["pk"]
 
-            # admin and instructors can access all registrations from the same consumer site
+            filters["lti_id"] = user.token.payload.get("context_id")
+            filters["consumer_site"] = get_object_or_404(
+                ConsumerSite, pk=user.token.payload["consumer_site"]
+            )
```

Comment:
Same comment here, I don't really understand why you are fetching this data in the database ?

Reply:
If I assign directly the UUID I get an error value, ValueError: Cannot assign "'5ac6ec8b-9857-4ad8-.. maybe I'm missing something?